xbmc/platform/win32/WIN32Util.cpp

```diff
@@ -1196,7 +1196,38 @@ HDR_STATUS CWIN32Util::ToggleWindowsHDR(DXGI_MODE_DESC& modeDesc)
   HDR_STATUS status = HDR_STATUS::HDR_TOGGLE_FAILED;
 
 #ifdef TARGET_WINDOWS_STORE
-  // Not supported - not implemented yet
+  auto hdmi = HdmiDisplayInformation::GetForCurrentView();
+
+  if (!hdmi)
+    return status;
+
+  const auto current = hdmi.GetCurrentDisplayMode();
+
+  const bool needHDR = (current.ColorSpace() != HdmiDisplayColorSpace::BT2020);
+
+  for (const auto& mode : hdmi.GetSupportedDisplayModes())
+  {
+    if ((!needHDR || (needHDR && mode.ColorSpace() == HdmiDisplayColorSpace::BT2020)) &&
+        mode.ResolutionHeightInRawPixels() == current.ResolutionHeightInRawPixels() &&
+        mode.ResolutionWidthInRawPixels() == current.ResolutionWidthInRawPixels() &&
+        mode.StereoEnabled() == false &&
+        fabs(mode.RefreshRate() - current.RefreshRate()) <= 0.00001)
+    {
+      if (current.ColorSpace() == HdmiDisplayColorSpace::BT2020) // HDR is ON
+      {
+        CLog::LogF(LOGINFO, "Toggle Windows HDR Off (ON => OFF).");
+        if (Wait(hdmi.RequestSetCurrentDisplayModeAsync(mode, HdmiDisplayHdrOption::None)))
```

Comment:
Does this test the bool return result of RequestSetCurrentDisplayModeAsync()?
Or Wait() own success code?

Reply:
See definition of Wait here:
https://github.com/xbmc/xbmc/blob/031dca9bd69d4e4fa2826690e23eea884b7e3a05/xbmc/platform/win10/AsyncHelpers.h#L63-L79

`Wait` returns result of `winrt::IAsyncOperation<TResult>` that in case of `RequestSetCurrentDisplayModeAsync` is

`winrt::IAsyncOperation<bool>`  then is returned bool.

Reply:
ok